java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java

```diff
@@ -51,31 +68,75 @@ public Type getType() {
         return new CompletedFuture<File>(zipOutPath.toFile());
 	}
 	
-	public static Path pack(final Path folder) throws IOException {
+	public Path pack(final Path folder, String namespace, String name, String tkName) throws IOException, URISyntaxException {
 		Path zipFilePath = Paths.get(folder.toAbsolutePath().toString() + ".zip");
-		String folderName = folder.getFileName().toString();
-	    try (
-	            FileOutputStream fos = new FileOutputStream(zipFilePath.toFile());
-	            ZipOutputStream zos = new ZipOutputStream(fos)
-	    ) {
-	        Files.walkFileTree(folder, new SimpleFileVisitor<Path>() {
-	            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-	                zos.putNextEntry(new ZipEntry(folderName + "/" + folder.relativize(file).toString()));
-	                Files.copy(file, zos);
-	                zos.closeEntry();
-	                return FileVisitResult.CONTINUE;
-	            }
-
-	            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-	                zos.putNextEntry(new ZipEntry(folderName + "/" + folder.relativize(dir).toString() + "/"));
-	                zos.closeEntry();
-	                return FileVisitResult.CONTINUE;
-	            }
-	        });
-	    }
+		String workingDir = zipFilePath.getParent().toString();
+		// com.ibm.streamsx.topology/lib/com.ibm.streamsx.topology.jar
+		File jarLocation = new File(ZippedToolkitStreamsContext.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath());
+		// com.ibm.streamsx.topology/lib/
+		File libDir = new File(jarLocation.getParent());
+		// com.ibm.streamsx.topology
+		File topologyToolkit = new File(libDir.getParent());	
+		
+		// tkManifest is the list of toolkits contained in the archive
+		PrintWriter tkManifest = new PrintWriter("manifest_tk.txt", "UTF-8");
+		
+		// mainComposite is a string of the namespace and the main composite.
+		// This is used by the Makefile
+		PrintWriter mainComposite = new PrintWriter("main_composite.txt", "UTF-8");
+		
+		tkManifest.println(tkName);
+		tkManifest.println("com.ibm.streamsx.topology");
+		
+		mainComposite.print(namespace + "::" + name);
+		
+		tkManifest.close();
+		mainComposite.close();
+		
+		List<Path> paths = new ArrayList<Path>();
+		paths.add(Paths.get(topologyToolkit.getAbsolutePath()));
+		paths.add(Paths.get(workingDir + "/manifest_tk.txt"));
+		paths.add(Paths.get(workingDir + "/main_composite.txt"));
+		paths.add(Paths.get(topologyToolkit.getAbsolutePath() + "/opt/python/templates/common/Makefile"));
+		paths.add(folder);
+		
+		addAllToZippedArchive(paths, zipFilePath);	
+		JSONObject jso = new OrderedJSONObject();
+		super.deleteToolkit(new File(paths.get(1).toString()), jso);
```

Comment:
Why not just `new File(paths.get(1).toString()).delete()` - no need to use JSON then.